### PR TITLE
feat: Fix left navigation UI issues and filter navigations to be displayed - EXO-66384 - Meeds-io/MIPs#72

### PR DESCRIPTION
**Pull request by GouadriaHanen**

### CODE REVIEW

**Comment by exo-swf** [comment]
Your PR triggers too many exo-ci builds! Please finish your work and then, set your PR ready! Thank you

File: component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java [hunk 1]
```diff
@@ -1923,4 +1939,42 @@ public static String getTranslatedLabel(String fieldName, long siteId, Locale lo
...
+      userNode.setChildren(removeEmptyNavigations(userNode.getChildren()));
+    }
+    return userNodeList;
+  }
+
+  private static boolean hasUserNodePageChild(UserNodeRestEntity userNode) {
+    if (userNode.getChildren() == null || userNode.getChildren().isEmpty()) {
+      return false;
+    }
+    boolean hasPageChild = false;
```
Comment (azayati) [inline component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java]:
```suggestion
    boolean hasPageChildNode = false;
```

File: component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java [hunk 2]
```diff
@@ -1923,4 +1939,42 @@ public static String getTranslatedLabel(String fieldName, long siteId, Locale lo
...
+  private static Visibility[] convertVisibilities(List<String> visibilityNames) {
+    if (visibilityNames == null) {
+      return Visibility.values();
+    }
+    return visibilityNames.stream()
+            .map(visibilityName -> Visibility.valueOf(StringUtils.upperCase(visibilityName)))
+            .toList()
+            .toArray(new Visibility[0]);
+  }
+  private static List<UserNodeRestEntity> removeEmptyNavigations(List<UserNodeRestEntity> userNodeList) {
```
Comment (azayati) [inline component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java]:
```suggestion
  private static void removeGroupNodesWithoutPageChildNodes(List<UserNodeRestEntity> userNodeList) {
```
Add a comment to explain what this method does exactly.

File: component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java [hunk 3]
```diff
@@ -1791,29 +1798,17 @@ public static List<SiteEntity> buildSiteEntities(List<PortalConfig> sites,
   public static SiteEntity buildSiteEntity(PortalConfig site,
                                            HttpServletRequest request,
                                            boolean expandNavigations,
+                                           List<String> visibilityNames,
                                            boolean excludeEmptyNavigationSites,
+                                           boolean temporalCheck,
+                                           boolean excludeEmptyNavigation,
```
Comment (azayati) [inline component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java]:
```suggestion
                                           boolean excludeGroupNodesWithoutPageChildNodes,
```

File: component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java [hunk 4]
```diff
@@ -1923,4 +1939,42 @@ public static String getTranslatedLabel(String fieldName, long siteId, Locale lo
...
+      if (userNode.getPageKey() == null && !hasUserNodePageChild(userNode)) {
+        i.remove();
+        continue;
+      }
+      userNode.setChildren(removeEmptyNavigations(userNode.getChildren()));
+    }
+    return userNodeList;
+  }
+
+  private static boolean hasUserNodePageChild(UserNodeRestEntity userNode) {
```
Comment (azayati) [inline component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java]:
```suggestion
  private static boolean hasPageChildNode(UserNodeRestEntity userNode) {
```

File: component/service/src/main/java/org/exoplatform/social/rest/impl/site/SiteRest.java [hunk 5]
```diff
@@ -104,6 +105,18 @@ public Response getSites(@Context
...
+                           @QueryParam("visibility")
+                           List<String> visibilityNames,
+                           @Parameter(description = "to check , in expandNavigations case , the navigation nodes scheduling start and end dates")
+                           @DefaultValue("false")
+                           @QueryParam("temporalCheck")
+                           boolean temporalCheck,
+                           @Parameter(description = "to expand site navigations nodes")
+                           @DefaultValue("false")
+                           @QueryParam("excludeEmptyNavigations")
+                           boolean excludeEmptyNavigations,
```
Comment (azayati) [inline component/service/src/main/java/org/exoplatform/social/rest/impl/site/SiteRest.java]:
```suggestion
                           @Parameter(description = "to exclude group nodes without page child nodes")
                           @DefaultValue("false")
                           @QueryParam("excludeGroupNodesWithoutPageChildNodes")
                           boolean excludeGroupNodesWithoutPageChildNodes,
```

File: component/service/src/main/java/org/exoplatform/social/rest/impl/site/SiteRest.java [hunk 6]
```diff
@@ -104,6 +105,18 @@ public Response getSites(@Context
                            @DefaultValue("false")
                            @QueryParam("expandNavigations")
                            boolean expandNavigations,
+                           @Parameter(description = "Multivalued visibilities of navigation nodes to retrieve, possible values: DISPLAYED, HIDDEN, SYSTEM or TEMPORAL. If empty, all visibilities will be used.", required = false)
+                           @Schema(defaultValue = "All possible values combined")
+                           @QueryParam("visibility")
+                           List<String> visibilityNames,
+                           @Parameter(description = "to check , in expandNavigations case , the navigation nodes scheduling start and end dates")
```
Comment (azayati) [inline component/service/src/main/java/org/exoplatform/social/rest/impl/site/SiteRest.java]:
```suggestion
                           @Parameter(description = "to check, in expandNavigations case, the navigation nodes scheduling start and end dates")
```

File: component/service/src/main/java/org/exoplatform/social/rest/impl/site/SiteRest.java [hunk 7]
```diff
@@ -181,6 +197,18 @@ public Response getSiteById(@Context
                               @DefaultValue("false")
                               @QueryParam("expandNavigations")
                               boolean expandNavigations,
+                              @Parameter(description = "Multivalued visibilities of navigation nodes to retrieve, possible values: DISPLAYED, HIDDEN, SYSTEM or TEMPORAL. If empty, all visibilities will be used.", required = false)
+                              @Schema(defaultValue = "All possible values combined")
+                              @QueryParam("visibility")
+                              List<String> visibilityNames,
+                              @Parameter(description = "to check , in expandNavigations case , the navigation nodes scheduling start and end dates")
```
Comment (azayati) [inline component/service/src/main/java/org/exoplatform/social/rest/impl/site/SiteRest.java]: see above

File: webapp/portlet/src/main/webapp/vue-apps/common/js/SiteService.js [hunk 8]
```diff
@@ -1,19 +1,26 @@
...
     formData.append('excludedSiteType', excludedSiteType);
   }
-  
+
   if (excludedSiteName) {
     formData.append('excludedSiteName', excludedSiteName);
   }
   formData.append('lang', eXo.env.portal.language);
   formData.append('excludeEmptyNavigationSites', excludeEmptyNavigationSites);
+  formData.append('excludeEmptyNavigations', excludeEmptyNavigations);
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/vue-apps/common/js/SiteService.js]: see above

**Review by azayati — CHANGES_REQUESTED** [review]

**Pull request merged**